tap-version-14-specification.md

```diff
@@ -0,0 +1,741 @@
+    ok 6 - pinged gold
+    1..6
+not ok 2 - ping servers
+  ---
+  servers_offline:
+    - saphire
+    - quartz
+  ...
+```
+
+### Giving up
+
+This listing reports that a pile of tests are going to be run.
+However, the first test fails, reportedly because a connection to the
+database could not be established. The program decided that continuing
+was pointless and exited.
+
+```
+TAP version 14
+1..573
+not ok 1 - database handle
+Bail out! Couldn't connect to database.
+```
+
+This listing reports a pile of tests where a child test bails out.
+The Harness MAY continue to run subsequent tests, but in most cases,
+SHOULD bubble the bail out up to the parent test.
+
+```
+TAP version 14
+1..54
+    # Subtest: 00-setup.js
+    not ok 1 - database handle
+    Bail out! Couldn't connect to database.
+Bail out! Coulnd't connect to database.
+```
+
+### Skipping a few
+
+The following listing plans on running 5 tests. However, our program
+decided to not run tests 2 thru 5 at all. To properly report this, the
+tests are marked as being skipped.
+
+```
+TAP version 14
+1..5
+ok 1 - approved operating system
+# $^0 is solaris
+ok 2 - # SKIP no /sys directory
+ok 3 - # SKIP no /sys directory
+ok 4 - # SKIP no /sys directory
+ok 5 - # SKIP no /sys directory
+```
+
+### Skipping everything
+
+This listing shows that the entire listing is a skip. No tests were run.
+
+```
+TAP version 14
+1..0 # skip English-to-French translator isn't installed
+```
+
+## Got spare tuits?
+
+The following example reports that four tests are run and the last two
+tests failed. However, because the failing tests are marked as things
+to do later, they are considered successes. Thus, a harness should
+report this entire listing as a success.
+
+```
+TAP version 14
+1..1
+    # Subtest: child program
+    1..4
+    ok 1 - Creating test program
+    ok 2 - Test program runs, no error
+    not ok 3 - infinite loop # TODO halting problem unsolved
+    not ok 4 - infinite loop 2 # TODO halting problem unsolved
+ok 1 - child program
+```
+
+## Creative liberties
+
+This listing shows an alternate output where the test numbers aren't
+provided. The test also reports the state of a ficticious board game
+as a YAML block. Finally, the test count is reported at the end.
+
+```
+TAP version 14
+ok - created Board
+ok
+ok
+ok
+ok
+ok
+ok
+ok
+  ---
+  message: "Board layout"
+  severity: comment
+  dump:
+     board:
+       - '      16G         05C        '
+       - '      G N C       C C G      '
+       - '        G           C  +     '
+       - '10C   01G         03C        '
+       - 'R N G G A G       C C C      '
+       - '  R     G           C  +     '
+       - '      01G   17C   00C        '
+       - '      G A G G N R R N R      '
+       - '        G     R     G        '
+  ...
+ok - board has 7 tiles + starter tile
+1..9
+```
+
+## CHANGES FROM TAP13
+
+- YAML blocks standardized to 2 space indentation
+- Subtests specified to behavior of `Test::More` and `node-tap`.
```

Comment:
Depending on how formal this specification is and how important the `Test::More` and `node-tap` references are, it might be relevant to include version numbers.